C++20 module does not compile with std.compat and GCC 15

When using `GCC 15.2.1`, the module compilation fails with the following error:

```bash
[18/21  71% :: 0.319] Building CXX object CMakeFiles/VulkanHppModule.dir/vulkan/vulkan.cppm.o
FAILED: [code=1] CMakeFiles/VulkanHppModule.dir/vulkan/vulkan.cppm.o CMakeFiles/VulkanHppModule.dir/vulkan_hpp.gcm 
/usr/bin/g++  -I/home/jan/repos/Vulkan-Hpp -I/home/jan/repos/Vulkan-Hpp/Vulkan-Headers/include -O3 -DNDEBUG -std=gnu++23 -MD -MT CMakeFiles/VulkanHppModule.dir/vulkan/vulkan.cppm.o -MF CMakeFiles/VulkanHppModule.dir/vulkan/vulkan.cppm.o.d -fmodules-ts -fmodule-mapper=CMakeFiles/VulkanHppModule.dir/vulkan/vulkan.cppm.o.modmap -MD -fdeps-format=p1689r5 -x c++ -o CMakeFiles/VulkanHppModule.dir/vulkan/vulkan.cppm.o -c /home/jan/repos/Vulkan-Hpp/vulkan/vulkan.cppm
In file included from /home/jan/repos/Vulkan-Hpp/Vulkan-Headers/include/vulkan/vk_platform.h:62,
                 from /home/jan/repos/Vulkan-Hpp/Vulkan-Headers/include/vulkan/vulkan.h:10,
                 from /home/jan/repos/Vulkan-Hpp/vulkan/vulkan.hpp:42,
                 from /home/jan/repos/Vulkan-Hpp/vulkan/vulkan.cppm:24:
/usr/lib/gcc/x86_64-pc-linux-gnu/15.2.1/include/stddef.h:451:3: error: conflicting declaration ‘typedef struct max_align_t max_align_t’
  451 | } max_align_t;
      |   ^~~~~~~~~~~
In file included from /usr/include/c++/15.2.1/cstddef:52,
                 from /usr/include/c++/15.2.1/x86_64-pc-linux-gnu/bits/stdc++.h:40,
                 from /usr/include/c++/15.2.1/bits/std.cc:30,
of module std, imported at /usr/include/c++/15.2.1/bits/std.compat.cc:30,
of module std.compat, imported at /home/jan/repos/Vulkan-Hpp/vulkan/vulkan.cppm:32:
/usr/lib/gcc/x86_64-pc-linux-gnu/15.2.1/include/stddef.h:451:3: note: previous declaration as ‘typedef struct max_align_t max_align_t’
  451 | } max_align_t;
      |   ^~~~~~~~~~~
/usr/include/bits/stdint-intn.h:24: confused by earlier errors, bailing out
```

it seems to accidentally be importing some std symbols twice. Using `std` instead of `std.compat` seems to fix this issue.
Should we change to it? We don't use `compat` either way and therefore shouldn't force the import on users imo.

--- a/snippets/CppmTemplate.hpp
+++ b/snippets/CppmTemplate.hpp
@@ -24,7 +24,7 @@ module;
 #include <vulkan/${api}_shared.hpp>
 
 export module ${api}_hpp;
-export import std.compat;
+export import std;
 
 export namespace VULKAN_HPP_NAMESPACE
 {

--- a/snippets/includes.hpp
+++ b/snippets/includes.hpp
@@ -28,7 +28,7 @@
 #else
 #  include <cassert>
 #  include <cstring>
-import std.compat;
+import std;
 #endif
 
 #if VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL == 1

--- a/vulkan/vulkan.cppm
+++ b/vulkan/vulkan.cppm
@@ -29,7 +29,7 @@ VULKAN_HPP_COMPILE_WARNING( VULKAN_HPP_CXX_MODULE_EXPERIMENTAL_WARNING )
 #include <vulkan/vulkan_shared.hpp>
 
 export module vulkan_hpp;
-export import std.compat;
+export import std;
 
 export namespace VULKAN_HPP_NAMESPACE
 {

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -38,7 +38,7 @@
 #else
 #  include <cassert>
 #  include <cstring>
-import std.compat;
+import std;
 #endif
 
 #if VULKAN_HPP_ENABLE_DYNAMIC_LOADER_TOOL == 1

